File: cpp/leetcode/053.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        if(nums.empty()) {
            return 0;
        }
        int maxSum = nums[0], curMax = nums[0];
        for(int i = 1; i < nums.size(); ++i){
            curMax = max(nums[i], nums[i] + curMax);
            maxSum = max(maxSum, curMax);
        }
        return maxSum;
    }
};

File: python/leetcode/053.py
class Solution(object):
    def maxSubArray(self, nums):
        if not nums:
            return 0;
        res, cur = nums[0], nums[0]
        for i in range(1, len(nums)):
            cur = max(nums[i], num[i] + cur)
            res = max(res, cur)
        return res